datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -136,6 +140,40 @@ private int getSourcePartitionFromEvent(BrooklinEnvelope event) {
         event.getMetadata().getOrDefault(BrooklinEnvelopeMetadataConstants.SOURCE_PARTITION, "-1"));
   }
 
+  @Override
+  public void broadcast(String destinationUri, DatastreamProducerRecord record,
+      SendCallback onBroadcastComplete, SendCallback onEventComplete) {
```

Comment:
With Future the responsibility will call on the caller (i.e. on the connector through EventProducer since most connectors call into TP through EventProducer), which will make connector logic (or probably EventProducer) more complicated where it will have to call this Future object. Probably in a separate thread so as to not make the send/broadcast blocking. 

Right now the most we do in callback is update metrics (in EventProducer) or log errors. Which makes it a good candidate for it being as a callback instead of Future. But thats just my understanding. 

In the current change above, broadcast callback is invoked when attempt to send to all partitions complete i.e. we call send for the record for each partition and in the broadcast callback we pass information of the total number of partitions and partitions to which send succeeded. However we dont wait for the send callback itself to complete.
onEventCompelete callback is the callback equivalent in send method. It is actually passed as the callback to send when we call send from broadcast.